client/src/amplify-ui/block-switcher/block-switcher.tsx

```diff
@@ -23,22 +23,36 @@ export class AmplifyBlockSwitcher {
   @Prop() readonly selectedTabHeadings: SelectedTabHeadings;
   /** tack on a new tab heading at highest priority */
   @Prop() readonly setNewSelectedTabHeadings: SetNewSelectedTabHeadings;
+  /** increments whenever the platform changes and we need to refresh the tabHeadings */
+  @Prop() readonly alwaysRerenderBlockSwitcher: number;
 
   @State() activeChildI = 0;
 
-  tabHeadings: string[] = [];
+  /** is a state so that it forces a rerender of the component */
+  @State() tabHeadings: string[] = [];
 
-  componentWillLoad() {
+  @Watch("alwaysRerenderBlockSwitcher")
+  triggerGatherHeadings() {
+    // just calling this.gather() is too fast -- we need to wait for the DOM to attach
```

Comment:
Can we validate that within one of [Stencil's Lifecycle Methods](https://stenciljs.com/docs/component-lifecycle) that we can't use one of these prior to doing a `setTimeout()`?